### PR TITLE
[Astro] Do not restart but reconfigure discovery service with new config

### PR DESCRIPTION
**Pull request by triller-telekom**
See https://github.com/eclipse/smarthome/pull/4079#issuecomment-359023592

Signed-off-by: Stefan Triller <stefan.triller@telekom.de>

### CODE REVIEW

**Comment by triller-telekom** [comment]
close and reopen for travis restart due to strange dmx binding tests failure (could not open a socket...)

File: extensions/binding/org.eclipse.smarthome.binding.astro/src/main/java/org/eclipse/smarthome/binding/astro/internal/discovery/AstroDiscoveryService.java [hunk 1]
```diff
@@ -61,15 +64,23 @@ public AstroDiscoveryService() {
     }
 
     @Override
+    @Activate
```
Comment (kaikreuzer) [inline extensions/binding/org.eclipse.smarthome.binding.astro/src/main/java/org/eclipse/smarthome/binding/astro/internal/discovery/AstroDiscoveryService.java]: You do not need activate and deactivate annotations, it is only required for modified.

**Comment by triller-telekom** [comment]
I thought it would make it consistent, but since we also keep the defaults for the parameters on `@Component` and `@Reference` without explicitly mentioning them, I think it is OK here to also only write what is required.

I have updated the PR.

File: extensions/binding/org.eclipse.smarthome.binding.astro/src/main/java/org/eclipse/smarthome/binding/astro/internal/discovery/AstroDiscoveryService.java [hunk 2]
```diff
@@ -66,10 +67,16 @@ protected void activate(Map<String, Object> configProperties) {
     }
 
     @Override
+    @Modified
     protected void modified(Map<String, Object> configProperties) {
         super.modified(configProperties);
     }
 
+    @Override
+    protected void deactivate() {
```
Comment (kaikreuzer) [inline extensions/binding/org.eclipse.smarthome.binding.astro/src/main/java/org/eclipse/smarthome/binding/astro/internal/discovery/AstroDiscoveryService.java]: As you aren't doing anything in here anymore, we do not have to keep this method as it does not carry any annotation anymore, right?

Comment (sjka) [reply extensions/binding/org.eclipse.smarthome.binding.astro/src/main/java/org/eclipse/smarthome/binding/astro/internal/discovery/AstroDiscoveryService.java]: hang on - `super` does! It's just the annotation which is missing here

**Comment by sjka** [comment]
> I thought it would make it consistent

I actually liked it 😞

**Comment by sjka** [comment]
Okay, as I just learned: we are doing it like that everwhere, so nevermind my comment. 
Then make it consistent with the activate, at least

**Comment by triller-telekom** [comment]
I have removed `activate` and `deactivate` because the OSGi framework will find them in the `super` class. Apparently only `modified` needs to be annotated in a child class to be found.

**Comment by triller-telekom** [comment]
The methods are back again and I will also update the documentation in a separate PR. It makes sense to see all DS methods in the implementing class and not have them hidden in a super class.

**Pull request merged**